src/trie.rs

```diff
@@ -124,4 +124,32 @@ mod tests {
 		assert!(tree.contains("foo.example.com", true));
 		assert!(!tree.contains("foo.example.com", false));
 	}
+
+	#[cfg(nightly)]
+	mod bench {
+		use super::*;
+		use std::fs;
+		use test::Bencher;
+
+		#[bench]
+		fn create_trie(b: &mut Bencher) {
+			let mut trie = Trie::new();
+			let path = concat!(env!("CARGO_MANIFEST_DIR"), "/bench/domains.txt");
+			let raw_list = fs::read_to_string(path).unwrap();
+			let list = crate::parser::Blocklist::parse(path, &raw_list)
+				.ok()
+				.unwrap();
+			drop(raw_list);
+			let domains: Vec<String> = list
+				.entries
+				.iter()
+				.map(|line| line.domain().0.clone())
+				.collect();
+			b.iter(|| {
+				for domain in &domains {
+					trie.insert(domain);
+				}
+			});
+		}
+	}
```

Comment:
I think the most important benchmark would be to check whether a domain is matched or not. We should add two benchmarks, one for the case that we expect the domain to be contained (_hit_), and one where the domain won't be contained (_miss_).

Reply:
hit is now impl